odp_sdk/tabular_storage_client.py

```diff
@@ -0,0 +1,394 @@
+import re
+from typing import Dict, List, Optional
+from uuid import UUID
+
+import requests
+from pydantic import BaseModel, PrivateAttr, field_validator
+
+from odp_sdk.dto import ResourceDto
+from odp_sdk.dto.table_spec import StageDataPoints, TableSpec
+from odp_sdk.dto.tabular_store import TableStage
+from odp_sdk.exc import OdpResourceExistsError, OdpResourceNotFoundError
+from odp_sdk.http_client import OdpHttpClient
+
+
+class OdpTabularStorageController(BaseModel):
+    _http_client: OdpHttpClient = PrivateAttr(default_factory=OdpHttpClient)
+    tabular_storage_endpoint: str = "/data"
+
+    @field_validator("tabular_storage_endpoint")
+    def endpoint_validator(self, v: str):
+        m = re.match(r"^/\w+(?<!/)", v)
+        if not m:
+            raise ValueError(f"Invalid endpoint: {v}")
+        return v
+
+    @property
+    def tabular_storage_url(self) -> str:
+        """
+        The URL of the tabular storage endpoint, including the base URL.
+
+        Returns:
+            The tabular storage URL
+        """
+        return f"{self._http_client.base_url}{self.tabular_storage_endpoint}"
+
+    def create_schema(self, resource_dto: ResourceDto, table_spec: TableSpec) -> TableSpec:
+        """
+        Create Schema
+
+        Args:
+            resource_dto: Dataset manifest
+            table_spec: Specifications of the schema that is being created
+
+        Returns:
+            Specifications of the schema that is being created
+
+        Raises
+            OdpResourceExistsError: If the schema already exists with the same identifier
+        """
+
+        if resource_dto.metadata.uuid:
+            url = f"{self.tabular_storage_url}/{resource_dto.metadata.uuid}/schema"
+        else:
+            url = f"{self.tabular_storage_url}/catalog.hubocean.io/dataset/{resource_dto.metadata.name}/schema"
```

Comment:
Could it be possible to move this check into an external function? There are a lot of checks like that throughout the class